src/main/java/seedu/binbash/storage/Storage.java

```diff
@@ -108,78 +177,101 @@ private ArrayList<String> readTxtFile() throws BinBashException, IOException {
-            throws InvalidCommandException, ParseException {
-        Command addCommand = addCommandParser.parse(commandArgs);
-        addCommand.execute(itemManager);
-    }
-
-    /**
-     * Handles the "setprofit" string by updating the profit-related properties of the most recently added item.
+     * Parses a single line from the text file to create an item object.
      *
-     * @param commandArgs The arguments of the setprofit command.
-     * @param itemManager The ItemList object containing the recently added item.
+     * @param line The string representation of a single line in the text file.
+     * @return The item object created from the line, or {@code null} if the item type is unknown.
      */
-    private void handleSetProfitString(String[] commandArgs, ItemList itemManager) {
-        List<Item> itemList = itemManager.getItemList();
-        Item recentlyAddedItem = itemList.get(itemList.size() - 1);
-
-        String totalUnitsPurchased = commandArgs[TOTAL_UNITS_PURCHASED_INDEX];
-        recentlyAddedItem.setTotalUnitsPurchased(Integer.parseInt(totalUnitsPurchased));
-
-        if (recentlyAddedItem instanceof RetailItem) {
-            String totalUnitsSold = commandArgs[TOTAL_UNITS_SOLD_INDEX];
-            ((RetailItem) recentlyAddedItem).setTotalUnitsSold(Integer.parseInt(totalUnitsSold));
+    protected Item convertLineToItem(String line) {
+        String[] itemElements = line.split("\\|");
+        String itemType = itemElements[ITEM_TYPE_INDEX];
+
+        switch (itemType) {
+        case "OperationalItem":
+
+            OperationalItem operationalItem = new OperationalItem(
+                    itemElements[ITEM_NAME_INDEX],
+                    itemElements[ITEM_DESCRIPTION_INDEX],
+                    Integer.parseInt(itemElements[ITEM_QUANTITY_INDEX]),
+                    Double.parseDouble(itemElements[ITEM_COST_PRICE_INDEX]),
+                    Integer.parseInt(itemElements[ITEM_THRESHOLD_INDEX])
+            );
+            operationalItem.setTotalUnitsPurchased(Integer.parseInt(itemElements[ITEM_TOTAL_UNITS_PURCHASED_INDEX]));
+
+            return operationalItem;
+        case "PerishableOperationalItem":
+
+            PerishableOperationalItem perishableOperationalItem = new PerishableOperationalItem(
+                    itemElements[ITEM_NAME_INDEX],
+                    itemElements[ITEM_DESCRIPTION_INDEX],
+                    Integer.parseInt(itemElements[ITEM_QUANTITY_INDEX]),
+                    LocalDate.parse(itemElements[ITEM_EXPIRATION_DATE_INDEX], EXPECTED_INPUT_DATE_FORMAT),
+                    Double.parseDouble(itemElements[ITEM_COST_PRICE_INDEX]),
+                    Integer.parseInt(itemElements[ITEM_THRESHOLD_INDEX])
+            );
+            perishableOperationalItem.setTotalUnitsPurchased(Integer.parseInt(
+                    itemElements[ITEM_TOTAL_UNITS_PURCHASED_INDEX]));
+
+            return perishableOperationalItem;
```

Comment:
And perhaps here, it can be refactored to:
```suggestion
        case "PerishableOperationalItem":
        
            PerishableOperationalItem perishableOperationalItem = new PerishableOperationalItem(
            itemName, 
            itemDescription, 
            itemQuantity,
            itemExpirationDate,
            itemCostPrice, 
            itemThreshold
            );
            operationalItem.setTotalUnitsPurchased(totalUnitsPurchased);
            
            return perishableOperationalItem;
```